### PR TITLE
refactor(test): replace custom testing utils with VTU

### LINKED ISSUE #574: Replace custom testing utils with vue test utils
We currently have a custom vue test utils because VTU didn't exist when I first started working on the router. It's time to change that.

This issue is up if anybody feels like doing the migration!

### PR DESCRIPTION
**Pull request by edison1105**
close #574

### CODE REVIEW

File: __tests__/RouterLink.spec.ts [hunk 1]
```diff
@@ -792,7 +795,7 @@ describe('RouterLink', () => {
       { to: locations.basic.string },
       locations.basic.normalized
     )
-    wrapper.find('a')!.click()
+    wrapper.find('a')!.trigger('click')
```
Comment (posva) [inline __tests__/RouterLink.spec.ts:798]: click should still work 🤔 
We can (and should) remove the `await nextTick()` and `await wrapper.find(...).click()`

File: __tests__/RouterView.spec.ts [hunk 2]
```diff
@@ -308,40 +311,44 @@ describe('RouterView', () => {
...
-      const wrapper = await mount(
+      const wrapper = mount(
         {
           template: `
         <div>
           <router-view/>
         </div>
         `,
-        },
+        } as any,
```
Comment (posva) [inline __tests__/RouterView.spec.ts:323]: Is the `as any` necessary? I should be okay without it

**Review by posva — CHANGES_REQUESTED** [review]

File: __tests__/RouterLink.spec.ts (hunk 1 above)
Comment (edison1105) [reply __tests__/RouterLink.spec.ts:798]: `.click()` is not work any more.

File: __tests__/RouterView.spec.ts (hunk 2 above)
Comment (edison1105) [reply __tests__/RouterView.spec.ts]: I removed it.

File: __tests__/RouterLink.spec.ts [hunk 3]
```diff
@@ -817,13 +818,14 @@ describe('RouterLink', () => {
 
   describe('v-slot', () => {
     const slotTemplate = `
+      <template #default="{route,href,isActive,isExactActive}">
```
Comment (posva) [inline __tests__/RouterLink.spec.ts:821]:
```suggestion
      <template #default="{ route, href, isActive, isExactActive} ">
```

File: __tests__/RouterLink.spec.ts [hunk 4]
```diff
@@ -817,13 +818,14 @@ describe('RouterLink', () => {
...
         <span>
           route: {{ JSON.stringify(route) }}
           href: "{{ href }}"
           isActive: "{{ isActive }}"
           isExactActive: "{{ isExactActive }}"
         </span>
-    `
+      </template>`
```
Comment (posva) [inline __tests__/RouterLink.spec.ts:828]:
```suggestion
      </template>
    `
```

File: __tests__/RouterView.spec.ts [hunk 5]
```diff
@@ -397,15 +414,18 @@ describe('RouterView', () => {
...
+        global: {
+          provide: route.provides,
+          components: { RouterView },
+        },
         slots: {
           default: `
-          <span>{{ route.name }}</span>
-          <component :is="Component"/>
-        `,
+            <template #default="{route,Component}">
```
Comment (posva) [inline __tests__/RouterView.spec.ts:425]:
```suggestion
            <template #default="{ route, Component }">
```

File: __tests__/RouterView.spec.ts [hunk 6]
```diff
@@ -424,16 +444,19 @@ describe('RouterView', () => {
...
+          provide: route.provides,
+          components: { RouterView },
+        },
         slots: {
           default: `
-          <keep-alive>
-            <component :is="Component"/>
-          </keep-alive>
-        `,
+          <template #default="{Component}">
```
Comment (posva) [inline __tests__/RouterView.spec.ts:455]:
```suggestion
          <template #default="{ Component }">
```

File: __tests__/guards/beforeRouteEnterCallback.spec.ts [hunk 7]
```diff
@@ -56,17 +56,21 @@ describe('beforeRouteEnter next callback', () => {
...
         template: `
       <div>
         <router-view/>
         <router-view name="other"/>
       </div>
       `,
-        components: { RouterView },
-      },
-      { router }
+      } as any,
```
Comment (posva) [inline __tests__/guards/beforeRouteEnterCallback.spec.ts:67]:
```suggestion
      },
```

**Review by posva — COMMENTED** [review]
Thanks a lot, this is looking good! I'm going to resolve the remaining comments and add a few changes locally

**Pull request merged**